### ISSUE TITLE
Add inbound connections to client pool

### ISSUE THREAD

**Issue by LLFourn** [comit-network/comit-rs#318]
`BamClientPool`: https://github.com/tenx-tech/swap/blob/3d3bd5816ddb42108baf291977de06f59dfbc417/application/comit_node/src/comit_client/bam.rs

Is meant to be the source of connections for other nodes. However if you get connected to by another party and then want to send them a message you will have to create a new connection because the existing one is not stored.

When the server receives a new connection it should add the client to `BamClientPool` under its `SocketAddress` and we should index the clients by socket address (for now). 

DoD:
- [x] `GET /peers` on Node2 returns Node1 network details once Node1 has sent a SWAP request to Node2
- [x] ~~`GET /peers` on Node1 and Node2 returns only node network details once Node2 has sent a SWAP request to Node1 (this goes after test above)~~

Blocked by #555 
Blocked by #620 
~~Blocked by Noise and Pubkeys (@LLFourn to open Noise ticket)~~

Note:

Ensuring that an old connection can be reused is out of scope for this ticket, because it requires connections to be uniquely identified.
